metals/src/main/scala/scala/meta/internal/metals/MultilineStringFormattingProvider.scala

```diff
@@ -114,56 +114,86 @@ final class MultilineStringFormattingProvider(
+        case (token: Constant.String, index: Int) =>
+          isMultilineString(sourceText, token) && hasStripMarginSuffix(
+            index,
+            tokens
+          ) && inToken(startPos, endPos, token)
+      }
+
+      def shouldFormatInterpolationString
+          : PartialFunction[(Token, Int), Boolean] = {
+        case (token: Interpolation.Start, index: Int)
+            if token.start < startPos.start => {
+          var endIndex = index + 1
           while (!tokens(endIndex)
               .isInstanceOf[Interpolation.End]) endIndex += 1
-          val end = tokens(endIndex)
-          stringFound = end.end > startPos.end
-          shouldAddPipes =
-            stringFound && isMultilineString(sourceText, start) &&
-              hasStripMarginSuffix(endIndex, tokens)
-        case _ =>
+          isMultilineString(sourceText, token) && hasStripMarginSuffix(
+            endIndex,
+            tokens
+          )
+        }
       }
-      tokenIndex += 1
-    }
-    shouldAddPipes
+      tokens.zipWithIndex
+        .exists(
+          shouldFormatMultiString orElse shouldFormatInterpolationString orElse {
+            case _ => false
+          }
+        )
+    } else false
   }
 
-  private def withToken(
-      textId: TextDocumentIdentifier,
-      range: Range,
-      newlineAdded: Boolean
-  )(fn: (String, meta.Position) => List[TextEdit]): Future[List[TextEdit]] =
-    Future {
-      val source = textId.getUri.toAbsolutePath
-      if (source.exists) {
-        val sourceText = buffer.get(source).getOrElse("")
-        val virtualFile = Input.VirtualFile(source.toString(), sourceText)
-        val startPos = range.getStart.toMeta(virtualFile)
-        val endPos = range.getEnd.toMeta(virtualFile)
-        if (pipeInScope(startPos, endPos, sourceText, newlineAdded)) {
-          val tokens =
-            virtualFile.tokenize.toOption
-          tokens
-            .filter(multilineStringInTokens(_, startPos, endPos, sourceText))
-            .map(_ => fn(sourceText, startPos))
-            .getOrElse(Nil)
-        } else Nil
-      } else Nil
-    }
+  private def withTokenOpt(textId: TextDocumentIdentifier, range: Range)(
+      fn: (
+          meta.Position,
+          meta.Position,
+          String,
+          Option[Tokens]
+      ) => List[TextEdit]
+  ): Future[List[TextEdit]] = {
+    val source = textId.getUri.toAbsolutePath
+    if (source.exists) {
+      val sourceText = buffer.get(source).getOrElse("")
+      val virtualFile = Input.VirtualFile(source.toString(), sourceText)
+      val startPos = range.getStart.toMeta(virtualFile)
+      val endPos = range.getEnd.toMeta(virtualFile)
+      Future { fn(startPos, endPos, sourceText, virtualFile.tokenize.toOption) }
+    } else Future.successful(Nil)
+  }
+  private def doubleQuoteNotClosed(
+      sourceText: String,
+      position: Position
+  ): Boolean = {
+    val splitLines = sourceText.split('\n')
```

Comment:
Could we maybe operate or indexes here?
Find the right line and then check `"` until the next `\n`?

It might be more efficient to not use split on the whole file. I am worried about the performance with large files.

Or alternatively, we could create a method `findLine` that would just do the substring on the needed line. What do you think?